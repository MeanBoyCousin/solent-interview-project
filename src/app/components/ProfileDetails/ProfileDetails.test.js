import React from 'react'
import 'dotenv/config'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { ProfileDetails } from './ProfileDetails'
import mockResponse from '../../../__mocks__/apiResponse.json'

afterEach(cleanup)

describe('ProfileDetails component', () => {
    test('should render with correct profile details', () => {
        const { asFragment } = render(
            <ProfileDetails profile={mockResponse.results[0]} />
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
