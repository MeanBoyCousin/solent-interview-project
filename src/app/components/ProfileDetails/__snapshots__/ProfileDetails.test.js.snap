// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileDetails component should render with correct profile details 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa dMxzUZ"
    style="opacity: 0.2; transform: translateY(30px) translateZ(0);"
  >
    <img
      alt="The profile picture for the member of staff you selected"
      class="profile"
      src="https://randomuser.me/api/portraits/women/55.jpg"
    />
    <h2>
      Nella Aalto
    </h2>
    <div>
      <span>
        <img
          alt="An icon of an envelope to symbolise a users email address"
          class="icon"
          src="/static/icons/email.svg"
        />
        <p>
          nella.aalto@example.com
        </p>
      </span>
      <span>
        <img
          alt="An icon of a crown to symbolise a users date of birth"
          class="icon"
          src="/static/icons/dob.svg"
        />
        <p>
          30/10/1952
        </p>
      </span>
      <span>
        <img
          alt="An icon of a phone to symbolise a users phone number"
          class="icon"
          src="/static/icons/phone.svg"
        />
        <p>
          042-458-19-31
        </p>
      </span>
    </div>
    <button>
      Close
    </button>
  </div>
</DocumentFragment>
`;
